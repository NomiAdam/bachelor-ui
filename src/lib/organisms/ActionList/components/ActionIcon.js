import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { IoIosStar, IoIosStarOutline } from 'react-icons/lib/io';
import { darkTheme, lightTheme } from '../../../constants/theme';

const StyledIconWrapper = styled.div`
  width: 100%;
  min-height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  cursor: pointer;
`;

const StyledOutline = styled(IoIosStarOutline)`
  color: ${darkTheme.black};
  font-size: 1.7em;
`;

const StyledStar = styled(IoIosStar)`
  color: ${lightTheme.lightBlue};
  font-size: 1.7em;
`;

const ActionIcon = ({ isIconActive, handleIconClick }) => (
    <StyledIconWrapper onClick={handleIconClick}>
        { isIconActive ? <StyledStar/> : <StyledOutline/> }
    </StyledIconWrapper>
);

ActionIcon.propTypes = {
    /**
     * Boolean indicating whether icon is active on not
     */
    isIconActive: PropTypes.bool,
    /**
     * Function to be called upon icon click,
     */
    handleIconClick: PropTypes.func,
};

export { StyledIconWrapper, StyledStar, StyledOutline };
export default ActionIcon;
