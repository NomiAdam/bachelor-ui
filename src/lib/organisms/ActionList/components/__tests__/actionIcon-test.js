/* eslint-disable no-undef */
import React from 'react';
import ActionIcon, { StyledIconWrapper } from '../ActionIcon';
import {shallow} from 'enzyme';

describe('ActionIcon', () => {
    it('should be defined', () => {
        expect(ActionIcon).toBeDefined();
    });
    it('should be match the snapshot', () => {
        expect(shallow(<ActionIcon/>)).toMatchSnapshot();
    });
    it('should be match the snapshot when isIconActive', () => {
        expect(shallow(<ActionIcon isIconActive/>)).toMatchSnapshot();
    });
    it('should be of type StyledDateWrapper', () => {
        expect(shallow(<ActionIcon/>).type()).toBe(StyledIconWrapper);
    })
});
