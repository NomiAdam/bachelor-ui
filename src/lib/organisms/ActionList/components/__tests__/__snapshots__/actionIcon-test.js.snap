// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionIcon should be match the snapshot 1`] = `
<ActionIcon__StyledIconWrapper>
  <ActionIcon__StyledOutline />
</ActionIcon__StyledIconWrapper>
`;

exports[`ActionIcon should be match the snapshot when isIconActive 1`] = `
<ActionIcon__StyledIconWrapper>
  <ActionIcon__StyledStar />
</ActionIcon__StyledIconWrapper>
`;
